infrahub_sdk/schema/__init__.py

```diff
@@ -102,6 +110,23 @@ def validate_data_against_schema(self, schema: MainSchemaTypesAPI, data: dict) -
                     message=f"{key} is not a valid value for {identifier}",
                 )
 
+    def set_cache(self, schema: dict[str, Any] | SchemaRootAPI | BranchSchema, branch: str | None = None) -> None:
```

Comment:
I think `set_cache` should be a private method as it's meant for internal use.

Reply:
it's not meant for internal use, it's meant for user to use during unit tests